### PR TITLE
fuzz: test for too many open txs in a flow

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4125

Describe changes:
- fuzz : finds if a flow has too many open/live transactions
  
This is to avoid DOS by quadratic complexity for protocols looping over the whole list of transactions looking for a
specific transation with an identifier, so that a new PDU gets processed within that transaction.

Comes from #6863 and CI should already catch missing protocol

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 6103

**Comment by victorjulien** [comment]
Should this be marked "ready for review"?

**Comment by catenacyber** [comment]
> Should this be marked "ready for review"?

Yes, I pushed some commits to fix the NFS bug triggered by CIFuzz...

File: rust/src/nfs/nfs.rs [hunk 1]
```diff
@@ -359,6 +350,19 @@ impl NFSState {
...
         tx.id = self.tx_id;
+        if self.transactions.len() > unsafe { NFS_MAX_TX } {
+            // set at least one another transaction to the drop state
+            for tx_old in &mut self.transactions {
+                if !tx_old.request_done || !tx_old.response_done {
+                    tx_old.request_done = true;
+                    tx_old.response_done = true;
+                    tx_old.is_file_closed = true;
+                    tx_old.tx_data.set_event(NFSEvent::TooManyTransactions as u8);
+                    self.events += 1;
```
Comment (inashivb) [inline rust/src/nfs/nfs.rs:361]: Q: what to we keep the events count for?

Comment (catenacyber) [reply rust/src/nfs/nfs.rs:361]: It is indeed unused, added a new commit to remove it.

@jasonish is there some rust tool to detect this ?

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7005

File: rust/src/nfs/nfs.rs (hunk 1 above)
Comment (jasonish) [reply rust/src/nfs/nfs.rs:361]: Do you mean unused variables?  Unread vars generate a warning by default.  However, this is a read and a write of the var, so probably not.  Maybe clippy can?

Comment (catenacyber) [reply rust/src/nfs/nfs.rs:361]: Yes unused variables.
Oh, I see technically `+= 1` is a read and a write... But it is not so much of a read semantically...

Does clippy get run on CI somewhere ?

**Pull request closed without merging**